tests/utils/test_dummpy_input_generators.py

```diff
@@ -0,0 +1,170 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from contextlib import nullcontext
+from unittest import TestCase
+
+from transformers import AutoConfig
+
+from parameterized import parameterized
+
+
+TEXT_ENCODER_MODELS = {"distilbert": "distilbert-base-cased"}
+
+VISION_MODELS = {"resnet": "hf-internal-testing/tiny-random-resnet"}
+
+SEQ2SEQ_MODELS = {"t5": "t5-small"}
+
+AUDIO_MODELS = {"whisper": "openai/whisper-tiny.en"}
+
+DUMMY_SHAPES = {
+    "batch_size": [2, 4],
+    "sequence_length": [16, 18],
+    "num_choices": [4, 6],
+    # image
+    "num_channels": [3, 1],
+    "width": [224, 150],
+    "height": [224, 150],
+    # audio
+    "feature_size": [40, 10],
+    "nb_max_frames": [300, 500],
+    "audio_sequence_length": [16000, 8000],
+}
+
+import itertools
+from typing import Any, Dict, Iterable
+
+import torch
+from transformers import AutoConfig
+
+from optimum.utils import DummyAudioInputGenerator, DummyTextInputGenerator, DummyVisionInputGenerator
+from optimum.utils.normalized_config import NormalizedConfigManager
+
+
+def grid_parameters(parameters: Dict[str, Iterable[Any]]) -> Iterable[Dict[str, Any]]:
+    """
+    Generate an iterable over the grid of all combinations of parameters
+    """
+    for params in itertools.product(*parameters.values()):
+        yield list(params)
+
+
+class GenerateDummy(TestCase):
+    @parameterized.expand(
+        grid_parameters(
+            {
+                "model_name": TEXT_ENCODER_MODELS.values(),
+                "batch_size": DUMMY_SHAPES["batch_size"],
+                "num_choices": DUMMY_SHAPES["num_choices"],
+                "sequence_length": DUMMY_SHAPES["sequence_length"],
+            }
+        )
+    )
+    def test_text_models(self, model_name: str, batch_size: int, num_choices: int, sequence_length: int):
+        # isn't this very verbose?
+        config = AutoConfig.from_pretrained(model_name)
+        normalized_config_class = NormalizedConfigManager.get_normalized_config_class(config.model_type)
+        normalized_config = normalized_config_class(config)
```

Comment:
Yes you are right it is!
We can create a `from_pretrained` method to `NormalizedConfig` handling all of this!

Reply:
Ok let's do it in an other PR